Framework/Algorithms/src/MultipleScatteringCylinderAbsorption.cpp

```diff
@@ -182,13 +185,17 @@ void MultipleScatteringCylinderAbsorption::exec() {
       eventList.getWeights(y_vec);
       eventList.getWeightErrors(err_vec);
 
-      apply_msa_correction(tth_rad, radius, coeff1, coeff2, coeff3, tof_vec,
-                           y_vec, err_vec);
+      HistogramX tof(std::move(tof_vec));
+      HistogramY y(std::move(y_vec));
+      HistogramE err(std::move(err_vec));
+
+      apply_msa_correction(tth_rad, radius, coeff1, coeff2, coeff3, tof, y,
+                           err);
```

Comment:
These vectors are *not* X, Y, and E. They have nothing to do with histograms, I think we should keep the old code here.